lib/item_repository.rb

```diff
@@ -0,0 +1,59 @@
+require "item"
+
+class ItemRepository
+
+  def all
+    sql = 'SELECT id, name, unit_price, quantity FROM items;'
+    result_set = DatabaseConnection.exec_params(sql,[])
+
+    items = []
+    result_set.each do |smth|
+      item = Item.new
+      item.id = smth['id'].to_i
+      item.name = smth['name']
+      item.unit_price = smth['unit_price']
+      item.quantity = smth['quantity']
+      items << item
+    end
+
+    return items
+  end
+
+  def find(id)
+    sql = 'SELECT id, name, unit_price, quantity FROM items WHERE id = $1;'
+    params = [id]
+    result_set = DatabaseConnection.exec_params(sql,params)[0]
```

Comment:
Very nice and smart way to put result_set = ...[0] instead of creating a new variable, I wish I have thought about this too.